src/functions_framework/__init__.py

```diff
@@ -174,6 +174,19 @@ def create_app(target=None, source=None, signature_type=None):
     app = flask.Flask(target, template_folder=template_folder)
     app.config["MAX_CONTENT_LENGTH"] = MAX_CONTENT_LENGTH
 
+    # 6. Handle legacy GCF Python 3.7 behavior
+    if os.environ.get("ENTRY_POINT"):
+        os.environ["FUNCTION_TRIGGER_TYPE"] = signature_type
+        os.environ["FUNCTION_NAME"] = os.environ.get("K_SERVICE", target)
```

Comment:
I think this raises some questions about how the environment variables should be handled in the 3.7 buildpack runtime. This has the effect of setting some of them, which is good... but why these? Should we be setting [all of them](https://cloud.google.com/functions/docs/env-var#nodejs_6_nodejs_8_python_37_and_go_111)?

Furthermore, should we set `FUNCTION_TRIGGER_TYPE` and `FUNCTION_NAME` for 3.8 as well? It would make it easier for users to migrate from 3.7 to 3.8.

Reply:
The remaining env vars don't have an equivalent in newer GCF runtimes, so they have to be set outside of the functions framework/buildpack. 

For the second question, there was precedent in the nodejs8 -> nodejs10 transition, where the older env vars were dropped. To prevent carrying on the tech debt I think they should not be available in Python 3.8.

Reply:
I'll take this discussion elsewhere 🙂